pkg/storage/utils/metadata/cs3.go

```diff
@@ -161,6 +164,14 @@ func (cs3 *CS3) Upload(ctx context.Context, req UploadRequest) error {
 			IfMatch: req.IfMatchEtag,
 		}
 	}
+	if len(req.IfNoneMatch) > 0 {
+		if req.IfNoneMatch[0] == "*" {
+			ifuReq.Options = &provider.InitiateFileUploadRequest_IfNotExist{}
+		}
+		// else {
+		//   the http upload will carry all if-not-match etags
+		// }
```

Comment:
remove?

Reply:
I left it in to explain the missing else case ...